### PR TITLE
Add records to planned test cases

### PR DESCRIPTION
**Pull request by rodesai**
### Description 
This patch rearranges the planned test cases, and adds input/output records into the persisted
test spec. spec.json is now split into plan.json and spec.json. plan.json contains the query plan
and configs. spec.json contains everything else. This split ensures that we serialize the plan
using the exact same ObjectMapper as a real KSQL node. The spec.json needs a specially configured
ObjectMapper to serialize RecordNode correctly (deal with null keys and decimals)

To review, look through the first commit. The second is just the generated changes.

### CODE REVIEW

File: ksql-functional-tests/src/test/java/io/confluent/ksql/test/planned/TestCasePlan.java [hunk 1]
```diff
@@ -16,55 +16,80 @@
...
-  private final TestCasePlanNode node;
+  private final TestCaseSpecNode specNode;
+  private final TestCasePlanNode planNode;
   private final String topology;
 
   TestCasePlan(
       final String version,
       final long timestamp,
-      final List<KsqlPlan> plan,
+      final List<KsqlPlan> planNode,
```
Comment (big-andy-coates) [inline ksql-functional-tests/src/test/java/io/confluent/ksql/test/planned/TestCasePlan.java]: nit: 

```suggestion
      final List<KsqlPlan> planNodes,
```

File: ksql-functional-tests/src/test/java/io/confluent/ksql/test/planned/TestCasePlan.java [hunk 2]
```diff
@@ -16,55 +16,80 @@
...
+      final String topology
+  ) {
+    this.specNode = Objects.requireNonNull(specNode, "spec");
+    this.planNode = Objects.requireNonNull(planNode, "plan");
     this.topology = Objects.requireNonNull(topology, "topology");
   }
 
   public List<KsqlPlan> getPlan() {
-    return node.getPlan();
+    return planNode.getPlan();
```
Comment (big-andy-coates) [inline ksql-functional-tests/src/test/java/io/confluent/ksql/test/planned/TestCasePlan.java]: I'd inline all of these accessors that access either planNode or specNode. Calls can just use `getPlanNode().getPlan()` etc.

File: ksql-functional-tests/src/test/java/io/confluent/ksql/test/planned/TestCaseSpecNode.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      @JsonProperty("timestamp") final long timestamp,
+      @JsonProperty("schemas") final Map<String, String> schemas,
+      @JsonProperty("inputs") final List<RecordNode> inputs,
+      @JsonProperty("outputs") final List<RecordNode> outputs
+  ) {
+    this.version = Objects.requireNonNull(version, "version");
+    this.timestamp = timestamp;
+    this.schemas = Objects.requireNonNull(schemas, "schemas");
+    this.inputs = inputs; // Objects.requireNonNull(inputs, "inputs");
+    this.outputs = outputs; // Objects.requireNonNull(outputs, "outputs");
```
Comment (big-andy-coates) [inline ksql-functional-tests/src/test/java/io/confluent/ksql/test/planned/TestCaseSpecNode.java]: nit: commented out code.

If these can be null I'd recommend making them `Optional`.  (You can plug Java8 support into Jackson if the mapper you're using doesn't have this already).

Plus, i'd recommend taking an immutable copy of the input.

File: ksql-functional-tests/src/test/resources/historical_plans/array_-_GENERATE_SERIES/5.5.0_1581572083765/plan.json [hunk 4]
```diff
@@ -0,0 +1,144 @@
+{
+  "plan" : [ {
```
Comment (big-andy-coates) [inline ksql-functional-tests/src/test/resources/historical_plans/array_-_GENERATE_SERIES/5.5.0_1581572083765/plan.json]: nit:

```suggestion
  "plans" : [ {
```

File: ksql-functional-tests/src/test/resources/historical_plans/array_-_GENERATE_SERIES/5.5.0_1581572083765/plan.json [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+            "properties" : { }
+          },
+          "options" : [ ]
+        },
+        "topicName" : "OUTPUT"
+      },
+      "queryId" : "CSAS_OUTPUT_0"
+    }
+  } ],
+  "configs" : {
```
Comment (big-andy-coates) [inline ksql-functional-tests/src/test/resources/historical_plans/array_-_GENERATE_SERIES/5.5.0_1581572083765/plan.json]: out of interest, why do we include the configs?  do we include the configs in the details persisted to the command topic too?

Wondering, as I thought the plan contained all the information needed to build the query, so the config seems superfluous.

**Comment by rodesai** [comment]
I'm going to merge this to unblock @spena and fix the feedback in a follow-up.

**Pull request merged**